### ISSUE TITLE
Iterator and process.maxTickDepth

### ISSUE THREAD

**Issue by peakji** [Level/leveldown#327]
We are using [fast-future](https://github.com/kesla/fast-future) in [iterator.js](https://github.com/Level/leveldown/blob/master/iterator.js#L34), but `process.maxTickDepth` is removed since node 0.12/v4 [(changes)](https://github.com/nodejs/node/wiki/API-changes-between-v0.10-and-v4).

`process.maxTickDepth` is now `undefined`, so `LIMIT` is `NaN`:

```javascript
var LIMIT = process.maxTickDepth / 2   // NaN
```

`setImmediate` will never fire because comparison between number and `NaN` will always be `false`:

```javascript
if (count >= LIMIT) {
	global.setImmediate(callback)
	count = 0
} else
	process.nextTick(callback)
count++
```

Never tested but I think this might cause `RangeError: Maximum call stack size exceeded` in long running processes.

**Comment by juliangruber** [comment]
that indeed sounds like a serious issue. I don't understand though why it will never fire, looks like in this condition it will actually trigger the `process.nextTick`

**Comment by peakji** [comment]
Yes, it will only trigger `process.nextTick`, and that's the problem.

The idea of fast-future is to call `setImediate` periodically to "clear" the call stack generated by the faster `process.nextTick`, to avoid `RangeError: Maximum call stack size exceeded` in recursions.

Because we have `cache` in iterators, most of the `next` calls are not really async operations, but tail recursion. However, the default `highWaterMark` will prevent ridiculously huge `cache`, and give node a chance to breathe.

Everythings fine right now except some potential extreme cases (almost impossible), but it'll be better if we fix the `NaN` issue and let fast-future work as designed. A hard coded `LIMIT` is a simple workaround.

cc @kesla

**Comment by kesla** [comment]
@peakji yo, I'll fix in fast-future right now. Thanks for bringing up

**Comment by kesla** [comment]
Fixed in 1.0.2 of fast-future

**Comment by peakji** [comment]
@kesla Nice!
